components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -727,6 +767,12 @@ public ServiceProvider getServiceProviderByClientId(String clientId, String clie
 
         endTenantFlow();
 
+        for (ApplicationMgtListener listener : listeners) {
+            if (!listener.doPostGetApplicationByClientId(serviceProvider, clientId, clientType, tenantDomain)) {
+                return null;
+            }
+        }
+
```

Comment:
Invoking listeners must be the final thing inside the service method. Just before the return. Also it must be outside any try catch blocks so that we don't need to unnecessarily handle any exception again in service method throw from listener.